fhirpipe/load/fhirstore.py

```diff
@@ -65,14 +65,23 @@ def find_fhir_resource(resource_type, identifier):
         else None
     """
     db_client = get_mongo_client()[fhirpipe.global_config["fhirstore"]["database"]]
+
+    # NOTE For now, URI simply consists in the resource name, need discussion about this
+    resource_type = uri
+
     if resource_type not in fhir_ids:
+        # If URI not found, don't do anything
+        if resource_type not in db_client.list_collection_names():
+            return
+
         results = db_client[resource_type].find(
-            {"id": identifier}
+            {"identifier": {"$elemMatch": {"value": {"$exists": True}}}},
+            ["id", "identifier"],
         )
 
         fhir_ids[resource_type] = {}
         for r in results:
-            # r_identifier = r["identifier"][0]["value"]
-            fhir_ids[resource_type][identifier] = str(r["_id"])
+            for r_identifier in r["identifier"]:
+                fhir_ids[resource_type][r_identifier["value"]] = r["id"]
 
     return fhir_ids[resource_type].get(identifier)
```

Comment:
You can handle the case where no result is found: either explicit the error or keep the provided `uri` as is

Reply:
Or return None I suppose accordong the `bind reference`

Reply:
Here, None is being returned if no result is found (.get() on a dict returns None by default if no key matches)